torch/csrc/autograd/engine.cpp

```diff
@@ -236,17 +253,9 @@ auto Engine::thread_init(int device) -> void {
   //
   // NB: We need an array here since neither DeviceGuard nor OptionalDeviceGuard
   // are movable.
-  std::array<c10::OptionalDeviceGuard,
-             static_cast<size_t>(c10::DeviceType::COMPILE_TIME_MAX_DEVICE_TYPES)>
-      guards;  // Guards! Guards!
-  if (device != -1) {
-    for (size_t i = 0; i < static_cast<size_t>(c10::DeviceType::COMPILE_TIME_MAX_DEVICE_TYPES); i++) {
-      auto* impl = c10::impl::device_guard_impl_registry[i].load();
-      if (impl && device < impl->deviceCount()) {
-        guards[i].reset_device(at::Device(static_cast<c10::DeviceType>(i), device));
```

Comment:
What was this code for? Why is it safe to remove it now?

Reply:
This was for setting the Cuda device. I have changed it now to do it without using guards.